internal/cephfs/volumeoptions.go

```diff
@@ -299,7 +300,13 @@ func newVolumeOptionsFromVolID(ctx context.Context, volID string, volOpt, secret
 	}
 
 	volOptions.ProvisionVolume = true
-	volOptions.RootPath, err = getVolumeRootPathCeph(ctx, &volOptions, cr, volumeID(vid.FsSubvolName))
+	info, err = getSubVolumeInfo(ctx, &volOptions, cr, volumeID(vid.FsSubvolName))
+	volOptions.RootPath = info.Path
+	if err != nil {
+		if errors.Is(err, ErrInvalidCommand) {
+			volOptions.RootPath, err = getVolumeRootPathCeph(ctx, &volOptions, cr, volumeID(vid.FsSubvolName))
+		}
+	}
```

Comment:
Store `info->Features` into `volOptions->Features`, such that in `checkSubvolumeHasFeature` function, you can invoke `getSubVolumeInfo` in case `Features` is not populated. Saves a call to ceph in the workflow.

Reply:
I suggested the same somehow I missed this one https://github.com/ceph/ceph-csi/pull/1458#issuecomment-688729196 :facepalm:

Reply:
Thanks Shyam, I'll make the changes.